src/Service/WorkloadReportService.php

```diff
@@ -65,24 +67,32 @@ public function getWorkloadReport(PeriodTypeEnum $viewPeriodType = PeriodTypeEnu
 
                 $worklogs = $this->getWorklogs($viewMode, $workerIdentifier, $dateFrom, $dateTo);
 
-                // Tally up logged hours in gathered worklogs for current period.
+                // Tally up logged hours in gathered worklogs for current period
                 $loggedHours = 0;
                 foreach ($worklogs as $worklog) {
                     $loggedHours += ($worklog->getTimeSpentSeconds() / 60 / 60);
                 }
 
                 $workerWorkload = $worker->getWorkload();
-
                 if (!$workerWorkload) {
                     $workerId = $worker->getUserIdentifier();
                     throw new \Exception("Workload of worker: $workerId cannot be unset when generating workload report.");
                 }
-                // Get total logged percentage based on weekly workload.
+
                 $roundedLoggedPercentage = $this->getRoundedLoggedPercentage($loggedHours, $workerWorkload, $viewPeriodType, $dateFrom, $dateTo);
 
+                // Count up total logged percentage to calculate average.
+                if (!$currentPeriodReached || 0 == $count) {
```

Comment:
This is not the correct way to calculate an average through the year. You cannot just sum the percentages and divide by number of months...

You should do:

```
$workloadSum = 0;
$loggedHoursSum = 0;

foreach ($periods as $period) {
  ...
  if (!$currentPeriodReached || 0 == $count) {
    $workloadSum += $workerWorklog;
    $loggedHoursSum += $loggedHours;
  }
} 
...

$workloadReportWorker->average = round($loggedHoursSum / $workloadSum, 2);
```